packages/react/src/__tests__/ReactElementJSX-test.js

```diff
@@ -306,8 +327,14 @@ describe('ReactElement.jsx', () => {
         });
       }
     }
-    // TODO: an explicit expect for no warning?
-    ReactDOM.render(JSXRuntime.jsx(Parent, {}), container);
+
+    const root = ReactDOMClient.createRoot(container);
+    await act(() => {
+      root.render(JSXRuntime.jsx(Parent, {}));
+    });
+
+    // Test shouldn't throw any errors.
+    expect(true).toBe(true);
```

Comment:
I couldn't figure out how to wrap the async stuff in `expect` to do something like:

```js
await expect(async () => {
  await act(() => {});
}).resolves.not.toThrow();
```

So we'll add this line instead, which includes the assertion but if the above throws the test will fail.

Something that would be nice to have is a `.not` operator for `.toWarnDev`:

```js
expect(whatever).not.toWarnDev()
```

Reply:
Any uncaught warning would fail the test right? I think that comment is sufficient.

Reply:
Yeah but we would want to fail on any new uncaught warnings right?

Reply:
Yeah good point, it would be nice to not have that the implicit assertion in the test though or having to expect true is true like I'm doing here